src/wallet/wallet.cpp

```diff
@@ -2517,6 +2517,7 @@ bool CWallet::CreateTransaction(interfaces::Chain::Lock& locked_chain, const std
             // TODO: pass in scriptChange instead of reservedest so
             // change transaction isn't always pay-to-bitcoin-address
             CScript scriptChange;
+            assert(scriptChange.empty());
```

Comment:
This seems unnecessary? How could it ever NOT be empty here?

Reply:
Removed. I added a potentially more useful assert below `scriptChange = GetScriptForDestination(dest);`